crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -649,3 +670,37 @@ impl<'a> Iterator for ErrorSources<'a> {
         current
     }
 }
+
+mod futures_helper {
```

Comment:
Can you move this to `bevy_utils::futures`? Seems like a reasonably generic tool.